### DISCUSSION TITLE
Entering Greek characters in a symbolic-input question

### DISCUSSION THREAD

**Discussion by jake-bobowski** [PrairieLearn/PrairieLearn#7908] [General]
I'm sorry if this is not the right place for this question.  However, I'd like to know if there is support or a recommended strategy for having students enter Greek characters into the answer box of a symbolic-input question.   I will be teaching an electromagnetism course and a lot of Greek characters appear.  Things like, $\epsilon_0$, $\mu_0$, $\rho$, $\sigma$, $\lambda$, ...

My current strategy is to write the characters in Unicode within the question text and then ask students to copy and paste these characters into the answer box when needed.  This works, but is a little inelegant.  Are there approaches?

**Comment by firasm** [comment]
This is how I'm doing it at the moment... but this is also not ideal. I am providing a table for the students to use that references the latex symbol with the value I want them to use. My hope is perhaps I'll teach them a little bit of LaTeX syntax.

<img width="1262" alt="Screen Shot 2021-06-25 at 9 30 58 PM" src="https://user-images.githubusercontent.com/2507459/123501889-fc854780-d5fc-11eb-912f-4651c5e763d9.png">

Eventually, a more ideal solution for symbolic questions, would be an equation editor:

<img width="766" alt="Screen Shot 2021-06-25 at 9 37 07 PM" src="https://user-images.githubusercontent.com/2507459/123501970-8e8d5000-d5fd-11eb-8d21-20358a374206.png">

There are several options available...

**Reply by firasm** [reply]
@nritschel  is working on something much better than this that solves the same problem here: https://github.com/PrairieLearn/PrairieLearn/pull/12651

Heads up @cbm755 , @jake-bobowski, Jonathan A!

**Comment by eliotwrobson** [comment]
@jake-bobowski names of greek letters used as variables get automatically converted to their latex counterparts by sympy (i.e. using `alpha` as a variable name will result in having this converted to the alpha latex symbol when shown back to the user). For entry this is a bit awkward, but students should be able to see what their answer is parsed into pretty easily.

I'm not sure if there's a nicer way of doing this outside of integrating a full-fledged equation editor (I'm not even sure how helpful this is in the scheme of things, depending on how complex student input is). Since this is more of a "best practices" issue rather than a full-fledged feature request, I'm going to go ahead and convert this to a discussion. 

Some of these issues should be partially alleviated by the changes introduced here: https://github.com/PrairieLearn/PrairieLearn/pull/7020

**Reply by jake-bobowski** [reply]
Thanks for the update @eliotwrobson!  For now, I'll probably continue to ask students to copy and paste unicode characters into their solutions.
